gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -254,7 +291,7 @@ protected void removeToken(final String tokenId) throws UnknownTokenException {
 
   protected boolean hasRemainingRenewals(final String tokenId, long renewInterval) {
     // Is the current time + 30-second buffer + the renewal interval is less than the max lifetime for the token?
-    return ((System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(30) + renewInterval) < getMaxLifetime(tokenId));
+    return ((System.currentTimeMillis() + 30000 + renewInterval) < getMaxLifetime(tokenId));
```

Comment:
Why is this 30 seconds buffer added?

Reply:
It is not added as part of this change, and I would have to revisit why it is there.

Reply:
The change to the literal value was unintentional